python/llm/test/inference_gpu/test_transformers_api_layernorm.py

```diff
@@ -0,0 +1,115 @@
+            input_ids = tokenizer.encode(PROMPT, return_tensors="pt").to(device)
+ 
+            model = Model.from_pretrained(model_path,
+                                        load_in_4bit=True,
+                                        optimize_model=False,
+                                        trust_remote_code=True)
+            model = model.to(device)
+            for layer_name, layer_module in model.named_modules():
+                if layer_name == layer_before_LayerNorm:
+                    layer_module.register_forward_hook(
+                        lambda module, input, output, layer_name=layer_name: pre_forward_hook(module, input,
+                                                                                            output, layer_name))
+                if layer_name == LayerNorm_layer:
+                    layer_module.register_forward_hook(
+                        lambda module, input, output, layer_name=layer_name: forward_hook(module, input,
+                                                                                        output, layer_name))
+            logits_base_model = (model(input_ids)).logits
+            # the list `layer_output` has only one element.
+            layer_tensor = self.layer_outputs.pop()
+            model.to('cpu')
+ 
+            opt_model = Model.from_pretrained(model_path,
+                                            load_in_4bit=True,
+                                            optimize_model=True,
+                                            trust_remote_code=True)
+            opt_model = opt_model.to(device)
+ 
+ 
+            def replace_forward_hook(module, input, output, layer_name):
+                output = self.pre_layer_outputs[0]
+                return output
+ 
+            for layer_name, layer_module in opt_model.named_modules():
+                if layer_name == layer_before_LayerNorm:
+                    layer_module.register_forward_hook(
+                        lambda module, input, output, layer_name=layer_name: replace_forward_hook(module, input,
+                                                                                                output, layer_name))
+                if layer_name == LayerNorm_layer:
+                    layer_module.register_forward_hook(
+                        lambda module, input, output, layer_name=layer_name: forward_hook(module, input,
+                                                                                        output, layer_name))
+            logits_optimized_model = (opt_model(input_ids)).logits
+            # the list `layer_output` has only one element.
+            opt_layer_tensor = self.layer_outputs[0]
+            opt_model.to('cpu')
+ 
+ 
+            LayerNorm_output_diff = []
+            for i, (t1, t2) in enumerate(zip(layer_tensor, opt_layer_tensor)):
+                if t1 is not None and t2 is not None:
+                    if isinstance(t1, torch.Tensor) and isinstance(t2, torch.Tensor):
```

Comment:
Can we remove these two lines?

Reply:
Deleting the NoneType checks may possibly cause errors according to the results of the last checks